packages/grid/x-grid/src/tests/reorder.XGrid.test.tsx

```diff
@@ -206,10 +215,151 @@ describe('<XGrid /> - Reorder', () => {
 
     render(<Test />);
     expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
-    const dragCol = getColumnHeaderCell(2).firstChild;
-    const dragEndEvent = createEvent.dragEnd(dragCol);
-    Object.defineProperty(dragEndEvent, 'dataTransfer', { value: { dropEffect: 'none' } });
+    const dragCol = getColumnHeaderCell(2).firstChild!;
+    const dragEndEvent = createDragEndEvent(dragCol, true);
     fireEvent(dragCol, dragEndEvent);
     expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
   });
+
+  describe('column disableReorder', () => {
+    it('should not allow to start dragging a column with disableReorder=true', () => {
+      let apiRef: GridApiRef;
+      const rows = [{ id: 0, brand: 'Nike' }];
+      const columns = [
+        { field: 'brand' },
+        { field: 'desc', disableReorder: true },
+        { field: 'type' },
+      ];
+
+      const Test = () => {
+        apiRef = useGridApiRef();
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid apiRef={apiRef} rows={rows} columns={columns} />
+          </div>
+        );
+      };
+
+      render(<Test />);
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand', 'desc', 'type']);
+      const dragCol = getColumnHeaderCell(1).firstChild!;
+      const targetCol = getColumnHeaderCell(0).firstChild!;
+
+      fireEvent.dragStart(dragCol);
+      fireEvent.dragEnter(targetCol);
```

Comment:
If the column can't be reordered we don't even allow the user to start dragging it. We could check if the `draggable` attribute is false or if `GRID_COLUMN_HEADER_DRAGGING_CSS_CLASS` was not added.